xs/sources/xsSnapshot.c

```diff
@@ -1545,6 +1527,120 @@ txSlot* fxUnprojectSlot(txMachine* the, txSnapshot* snapshot, txSlot* slot)
 	return slot;
 }
 
+int fxUseSnapshot(txMachine* the, txSnapshot* snapshot)
```

Comment:
Right, this must take a snapshot in RAM, and initializes a (blank) `txMachine` struct with the contents, for the "create new machine from the old one but within the same process" operation.

Reply:
No the `txSnapshot` is basically the wrapper around the `FILE` struct that `fxSnapshotRead` uses in `xsnap-worker.c`